src/openfermion/utils/_grid.py

```diff
@@ -13,80 +13,285 @@
 from __future__ import absolute_import
 
 import itertools
+import numpy
+import scipy
+import scipy.linalg
+
+
+# Exceptions.
+class OrbitalSpecificationError(Exception):
+    pass
 
 
 class Grid:
```

Comment:
Very good idea building out this Grid class to encapsulate some of the tricky reciprocal lattice stuff.